src/linearAlgebra/BlasLapackKernels.h

```diff
@@ -15,9 +15,10 @@ namespace dftefe
       /**
        * @brief namespace class for BlasLapack kernels not present in blaspp.
        */
-
-      template <typename ValueType, dftefe::utils::MemorySpace memorySpace>
-      class Kernels
+      template <typename ValueType1,
+                typename ValueType2,
+                dftefe::utils::MemorySpace memorySpace>
+      class KernelsTwoValueTypes
```

Comment:
@bikashkanungo @iancclin  I have used two different classes for one and two value type templates because C++ only allows partial specialization of the full template class, not function template partial specialization. For example if we do partial specialization of the full class for one value type, we have to separately instantiate for the two value type functions when both the value types are same vs when they are different (Hadamard product for example). That would be very confusing since the implementation is the same whether the value types are different or same. Partial specialization only makes sense when there is a clean fork in implementation for all member functions, like the memorySpace template parameter.